portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase

```diff
@@ -374,6 +374,63 @@ definition {
 		AssertElementPresent(locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_ERROR_NO_MAPPING_SELECTED");
 	}
 
+	@description = "LPS-123218 TC-2 Assertion for: first name cannot be used for user matching even if LDAP import is enabled."
+	@priority = "4"
+	test AssertFirstNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled {
+		property apacheds.enabled = "true";
+		property app.server.bundles.size = "0";
+		property test.name.skip.portal.instance = "SAML#AssertFirstNameCannotBeUsedForUserMatchingEvenWithLDAPEnabled";
+
+		PortalSettings.addLDAPServer(LDAPServerName = "ApacheDS Server");
+
+		PortalSettings.editLDAPServer(LDAPServerName = "ApacheDS Server");
+
+		PortalSettings.testLDAPConnection();
+
+		var enableSettingList = "enabled,ldapImportEnabled";
+
+		PortalSettings.configureLDAPSetting(enableSettingList = "${enableSettingList}");
+
+		PortalSettings.viewEnabledLDAPSetting(enableSettingList = "${enableSettingList}");
+
+		ApplicationsMenu.gotoPortlet(
+			category = "Security",
+			panel = "Control Panel",
+			portlet = "SAML Admin");
+
+		SAMLRole.configure(
+			samlEntityID = "samlsp",
+			samlRoleType = "Service Provider");
+
+		SAMLNavigation.gotoIdentityProviderConnections();
+
+		AssertClick(
+			locator1 = "CPSAMLAdminIdentityProviderConnection#ADD_IDENTITY_PROVIDER_BUTTON",
+			value1 = "Add Identity Provider");
+
+		Type(
+			locator1 = "TextInput#SAML_NAME_FIELD",
+			value1 = "samlidp");
+
+		Type(
+			locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_ENTITY_ID_FIELD",
+			value1 = "samlidp");
+
+		Click(locator1 = "CPSAMLAdminIdentityProviderConnection#IDENTITY_PROVIDER_CONNECTION_USER_RESOLUTION_ATTRIBUTE");
```

Comment:
SACRED: We could combine types and clicks from test case level, and put them into a dedicated macro